pkg/app/piped/cloudprovider/kubernetes/kubectl.go

```diff
@@ -181,3 +181,37 @@ func (c *Kubectl) Delete(ctx context.Context, kubeconfig, namespace string, r Re
 	}
 	return nil
 }
+
+func (c *Kubectl) Get(ctx context.Context, kubeconfig, namespace string, r ResourceKey) (m Manifest, err error) {
+	defer func() {
+		kubernetesmetrics.IncKubectlCallsCounter(
+			c.version,
+			kubernetesmetrics.LabelGetCommand,
+			err == nil,
+		)
+	}()
+
+	args := make([]string, 0, 7)
+	if kubeconfig != "" {
+		args = append(args, "--kubeconfig", kubeconfig)
+	}
+	if namespace != "" {
+		args = append(args, "--namespace", namespace)
+	}
+	args = append(args, "get", r.Kind, r.Name, "-o", "yaml")
+
+	cmd := exec.CommandContext(ctx, c.execPath, args...)
+	out, err := cmd.CombinedOutput()
+
+	if strings.Contains(string(out), "(NotFound)") {
+		return Manifest{}, fmt.Errorf("not found manifest %v, (%w), %v", r, ErrNotFound, err)
+	}
+	if err != nil {
+		return Manifest{}, fmt.Errorf("failed to get: %s, %v", string(out), err)
+	}
+	ms, err := ParseManifests(string(out))
+	if err != nil {
+		return Manifest{}, fmt.Errorf("failed to parse manifests %v: %v", r, err)
+	}
+	return ms[0], nil
```

Comment:
I think we need to check the existence since the list could be empty.

Reply:
If the resource doesn't exist, we can handle it by this line.
Are there any cases the resource exists but that manifest is empty?

Reply:
You are right.
Tbh I still think that we should have a check because the handle above is based on the string message and depending on the external tool.
But it is up to you.

Reply:
I see.
Let me fix it to be safe.